src/us/kbase/workspace/test/database/provenance/CreditMetadataTest.java

```diff
@@ -0,0 +1,1052 @@
+		}
+	}
+
+	@Test
+	public void buildMaximal() throws Exception {
+		for (Map.Entry<String, String> entry : VALID_PID_MAP.entrySet()) {
+			final CreditMetadata cm = CreditMetadata
+					.getBuilder(entry.getKey(), RESOURCE_TYPE_STRING,
+							CONTRIBUTOR_LIST, TITLE_LIST)
+					.withLicense(LICENSE_STRING_UNTRIMMED)
+					.withVersion(VERSION_STRING_UNTRIMMED)
+					.withComments(COMMENTS_LIST_UNTRIMMED)
+					.withDates(DATE_LIST)
+					.withFunding(FUNDING_LIST)
+					.withRelatedIdentifiers(RELATED_ID_LIST)
+					.build();
+			assertCreditMetadataFields(cm, entry.getValue(), RESOURCE_TYPE,
+					CONTRIBUTOR_LIST, TITLE_LIST, LICENSE_STRING, VERSION_STRING,
+					COMMENTS_LIST, DATE_LIST, FUNDING_LIST,
+					RELATED_ID_LIST);
+		}
+	}
+
+	@Test
+	public void buildMinimalOverwriteNullEmptyString() throws Exception {
+		for (final String nullOrWs : WHITESPACE_STRINGS_WITH_NULL) {
+			final CreditMetadata cm = CreditMetadata
+					.getBuilder(IDENTIFIER_STRING, RESOURCE_TYPE_STRING,
+							CONTRIBUTOR_LIST, TITLE_LIST)
+					.withLicense(LICENSE_STRING_UNTRIMMED)
+					.withLicense(nullOrWs)
+					.withVersion(VERSION_STRING_UNTRIMMED)
+					.withVersion(nullOrWs)
+					.withDates(DATE_LIST)
+					.build();
+			assertCreditMetadataFields(cm, IDENTIFIER_STRING, RESOURCE_TYPE,
+					CONTRIBUTOR_LIST, TITLE_LIST, null, null,
+					ELS, DATE_LIST, ELF, ELRI);
+		}
+	}
+
+	@Test
+	public void buildMaximalOverwriteNullEmptyLists() throws Exception {
+		final CreditMetadata cm = CreditMetadata
+				.getBuilder(IDENTIFIER_STRING, RESOURCE_TYPE_STRING,
+						CONTRIBUTOR_LIST, TITLE_LIST)
+				.withDates(DATE_LIST)
+				.withDates(Arrays.asList(null, null, null, null))
+				.withFunding(FUNDING_LIST)
+				.withFunding(null)
+				.withRelatedIdentifiers(RELATED_ID_LIST)
+				.withRelatedIdentifiers(ELRI)
+				.withVersion(VERSION_STRING_UNTRIMMED)
+				.build();
+		assertCreditMetadataFields(cm, IDENTIFIER_STRING, RESOURCE_TYPE,
+				CONTRIBUTOR_LIST, TITLE_LIST, null, VERSION_STRING);
+	}
+
+	@Test
+	public void buildMaximalDedupeLists() throws Exception {
+		final CreditMetadata cm = CreditMetadata
+				.getBuilder(IDENTIFIER_STRING,
+						RESOURCE_TYPE_STRING,
+						CONTRIBUTOR_LIST_DUPES_NULLS,
+						TITLE_LIST_DUPES_NULLS)
+				.withComments(COMMENTS_LIST_UNTRIMMED)
+				.withDates(DATE_LIST_DUPES_NULLS)
+				.withFunding(FUNDING_LIST_DUPES_NULLS)
+				.withRelatedIdentifiers(RELATED_ID_LIST_DUPES_NULLS)
+				.build();
+				assertCreditMetadataFields(cm, IDENTIFIER_STRING, RESOURCE_TYPE,
+				CONTRIBUTOR_LIST, TITLE_LIST, null, null,
+				COMMENTS_LIST, DATE_LIST, FUNDING_LIST,
+				RELATED_ID_LIST);
+	}
+
+	@Test
+	public void assertContributorsImmutable() throws Exception {
+		// same as CONTRIBUTOR_LIST
+		final List<Contributor> contributorList = new ArrayList<>(Arrays.asList(C1, C2));
+		final CreditMetadata cm = CreditMetadata
+				.getBuilder(IDENTIFIER_STRING, RESOURCE_TYPE_STRING,
+						CONTRIBUTOR_LIST, TITLE_LIST)
```

Comment:
```suggestion
						contributorList, TITLE_LIST)
```

Reply:
Ugh... this is what happens when you try to code after waking up at ridiculous o'clock. (I know you're in a permanent state of sleep deprivation, but I'm not as good at it as you are.)

Reply:
:+1: